automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppLifecycleSpec.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package apps
+
+import cats.effect.IO
+import org.broadinstitute.dsde.workbench.DoneCheckable
+import org.broadinstitute.dsde.workbench.google2.{streamFUntilDone, streamUntilDoneOrTimeout}
+import org.broadinstitute.dsde.workbench.leonardo.LeonardoApiClient.defaultCreateAppRequest
+import org.broadinstitute.dsde.workbench.leonardo.http.{ListAppResponse, PersistentDiskRequest}
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials, Uri}
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.tagobjects.Retryable
+import org.scalatest.{DoNotDiscover, ParallelTestExecution}
+
+import scala.concurrent.duration._
+
+@DoNotDiscover
+class AppLifecycleSpec
+    extends GPAllocFixtureSpec
+    with LeonardoTestUtils
+    with GPAllocUtils
+    with ParallelTestExecution
+    with TableDrivenPropertyChecks {
+  implicit val auth: Authorization =
+    Authorization(Credentials.Token(AuthScheme.Bearer, ronCreds.makeAuthToken().value))
+
+  override def withFixture(test: NoArgTest) =
+    if (isRetryable(test))
+      withRetry(super.withFixture(test))
+    else
+      super.withFixture(test)
+
+  val customCreateAppRequest = defaultCreateAppRequest.copy(
+    diskConfig = Some(
+      PersistentDiskRequest(
+        randomDiskName,
```

Comment:
IntelliJ suggests the parens, and I agree because otherwise it looks like we're using the same fixed value in two places that should be uniquely generated
```suggestion
        randomDiskName(),
```